vllm/spec_decode/top1_proposer.py

```diff
@@ -73,6 +73,29 @@ def get_proposals(
                 execute_model_req=nonzero_execute_model_req,
                 sample_len=proposal_len,
             )
+            if maybe_sampler_output is not None:
+                # Some sequences do not get speculative tokens from
+                # the draft worker. Remove these sequences from nonzero
+                # proposal len seqs to reduce scoring overhead
+                zero_seq_idxs = []
+                for seq_idx, sampler_output in zip(
+                        nonzero_proposal_len_indices, maybe_sampler_output):
+                    if sampler_output is None:
+                        proposal_lens[seq_idx] = 0
+                        zero_seq_idxs.append(seq_idx)
+                nonzero_proposal_len_indices = [
+                    idx for idx in nonzero_proposal_len_indices
+                    if idx not in zero_seq_idxs
```

Comment:
This is O(bs^2), suggest using `set` to make it O(bs)

Reply:
(there's an argument here that for small BS, set may take just as much time. not sure).

Reply:
It doesn't hurt to use set tho. Will do.